kclvm/runner/src/lib.rs

```diff
@@ -1,2 +1,53 @@
+use kclvm_ast::ast::Program;
+use kclvm_sema::resolver::scope::ProgramScope;
+use runner::{ExecProgramArgs, KclvmRunner, KclvmRunnerOptions};
+
+pub mod assembler;
 pub mod command;
+pub mod linker;
 pub mod runner;
+
+#[cfg(test)]
+pub mod tests;
+
+/// After the kcl program passed through kclvm-parser and kclvm-sema in the compiler frontend,
+/// KCLVM needs to generate corresponding LLVM IR, dylibs or executable file for kcl program
+/// in the compiler backend.
+///
+/// Method “execute” is the entry point for the compiler backend.
+///
+/// It returns the KCL program executing result as Result<a_json_string, an_err_string>,
+/// and mainly takes "program" (ast.Program returned by kclvm-parser) and "scope"
+/// (ProgramScope returned by kclvm-sema) as input.
+///
+/// "plugin_agent" is related to KCLVM plugin.
+/// "args" is the items selected by the user in the KCLVM CLI.
+///
+/// In the method, dylibs is generated by KclvmAssembler, and method "KclvmAssembler::gen_dylibs"
+/// will return dylibs path in a "Vec<String>";
+///
+/// After linking all dylibs by KclvmLinker, method "KclvmLinker::link_all_dylibs" will return a path
+/// for dylib.
+///
+/// At last, KclvmRunner will be constructed and call method "run" to execute the kcl program.
+pub fn execute(
```

Comment:
Add some Samples directly in the doc comment for testing using the ``` code block. For example:
```rust
use kclvm::runner::execute;
// Omit other usages

fn main() {
    let result = execute(..).unwrap();
    assert_eq!(result, ...);
}
```